code/API_definitions/connected-network-type.yaml

```diff
@@ -0,0 +1,416 @@
+      - `2G`, if device is connected to the 2G network
+      - `3G`, if device is connected to the 3G network
+      - `4GLTE`, if device is connected to the 4G network
+      - `5GNSA`, if device's is connected to the 4G network but using 5G New Radio (NR) capability
+      - `5GSA`, if device is connected to the 5G network
+      - `NON3GPP4G`, if device is connected to the 4G network via WiFi network
+      - `NON3GPP5G`, if device's is connected to the 5G network via WiFi network
+
+
+    * **LastStatusTime** : This property specifies the time when the status was last updated. Its presence in the response indicates the freshness of the information, while its absence implies the information may be outdated or its freshness is uncertain.
+
+    # API Functionality
+
+    The API exposes following capabilities:
+
+    ## Connected Network Type
+
+    The endpoint `POST /retrieve-network-type` allows to get connected Network Type.
+
+
+    ## Further info and support
+
+    (FAQs will be added in a later version of the documentation)
+
+  termsOfService: http://swagger.io/terms/
+  contact:
+    email: project-email@sample.com
+  license:
+    name: Apache 2.0
+    url: https://www.apache.org/licenses/LICENSE-2.0.html
+  version: wip
+externalDocs:
+  description: Product documentation at CAMARA
+  url: https://github.com/camaraproject/
+
+servers:
+  - url: "{apiRoot}/network-type-retrieval/v0"
+    variables:
+      apiRoot:
+        default: http://localhost:9091
+        description: API root
+tags:
+  - name: Connected  Network Type
+    description: Operations to get the network type device is connected to
+paths:
+  /retrieve-network-type:
+    post:
+      tags:
+        - Connected network type
+      summary: "Get the connected network type"
+      description: Get the connected network type
+      operationId: getConnectedNetworkType
+      parameters:
+        - $ref: '#/components/parameters/x-correlator'
+      security:
+        - openId:
+            - connected-network-type:read
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/RequestConnectedNetworkType"
+        required: true
+      responses:
+        "200":
+          description: Contains connected network type
+          headers:
+            x-correlator:
+              $ref: '#/components/headers/x-correlator'
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ConnectedNetworkTypeResponse"
+              examples:
+                Connected-to-4G:
+                  value:
+                    lastStatusTime: "2024-02-20T10:41:38.657Z"
+                    connectedNetworkType: 4G
```

Comment:
4G you have not used here inside your enum:

```yaml
  enum:
        - 2G
        - 3G
        - 4GLTE
        - 5GNSA
        - 5GSA
        - NON3GPP4G
        - NON3GPP5G
        - NOT-CONNECTED
```

Maybe you wanted to use `4GLTE` ?

Reply:
We are hearing of one use case more regularly "Companies want to check if a number is a VoIP number vs a nonVoIP number" , any chance we could get that as one of the options here?